### PR TITLE
[4.0] Fix query after prepared statements

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #34417.

### Summary of Changes
Fix binding in sql statement when ordering with title.

Thank you for help, @joomdonation and @richard67 

### Testing Instructions
See issue #34417.
Make a module tags-popular and select ordering "title"

### Actual result BEFORE applying this Pull Request
Error 

### Expected result AFTER applying this Pull Request
Correct output of tags ordered by title

### CODE REVIEW

File: modules/mod_tags_popular/src/Helper/TagsPopularHelper.php [hunk 1]
```diff
@@ -142,13 +140,34 @@ public static function getList(&$params)
 					->order($db->quoteName('a.title') . ' ' . $order_direction);
 
 				$query = $equery;
+
+				// Re-bind value for t.access
```
Comment (richard67) [inline modules/mod_tags_popular/src/Helper/TagsPopularHelper.php]:
```suggestion
				// Rebind value for t.access
```
Hmm, here "Re-bind" and in comments below "Rebind". It should be consistent, I think. @brianteeman Which one is right?

Comment (brianteeman) [reply modules/mod_tags_popular/src/Helper/TagsPopularHelper.php]: rebind

Comment (chmst) [reply modules/mod_tags_popular/src/Helper/TagsPopularHelper.php]: thx

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 7abb490969a07377d1fe49330795123493b698f4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 7abb490969a07377d1fe49330795123493b698f4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by joomdonation** [comment]
@chmst Looking more at `DatabaseQuery` class, I found a better solution and made PR https://github.com/chmst/joomla-cms/pull/12 to your branch. Could you please review, test and if it works good, merge it please.

The current code require us to rebind value for each parameters in the right order (depend on the position it is called in the query), so it is not good and flexible enough (don't know how to explain more clear, happy to discuss directly on Glip if needed)

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 01edbaa5ce794e6beec8b52f2e358a636800d7e8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by richard67** [comment]
Back to pending. One more test needed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by richard67** [comment]
@joomdonation It's not sufficient just to remove the RTC label here on GitHub because the issue tracker still has status "Ready to Commit" and so will add it back with the next change to the issue, e.g. if someone tests again. For removing RTC you have to change the status in the issue tracker back to pending with some comment at the bottom, same way as when setting RTC.

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 01edbaa5ce794e6beec8b52f2e358a636800d7e8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by joomdonation** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34422">issues.joomla.org/tracker/joomla-cms/34422</a>.</sub>

**Comment by richard67** [comment]
Thanks!

**Pull request merged**